### PR TITLE
Upstream sync setup

### PR DESCRIPTION
**Pull request by ced0ps**

### CODE REVIEW

**Review by ubergesundheit — APPROVED** [review]
Changes look good!

Did you do this upgrade using the git subtree method? Didn't expect that many commits from upstream. :sweat_smile:

**Comment by ubergesundheit** [comment]
Please wait until you merge this into master! I want to fix something

**Review by yasn77 — APPROVED** [review]
TBH this PR is too big to properly review... But I trust you @ced0ps 👍🏽 

:shipit:

**Pull request merged**
